packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateFormV2.test.tsx

```diff
@@ -0,0 +1,679 @@
+    } = await commonSubmitSetup({
+      stripeOverride: {
+        ...defaultStripeOverride(),
+        confirmCardSetup: jest.fn().mockResolvedValue({ setupIntent: null }),
+      },
+    });
+
+    expect(screen.queryByTestId('error-message-container')).toBeInTheDocument();
+    expect(screen.getByText('basic-error-message')).toBeInTheDocument();
+    expect(screen.queryByTestId('paymentForm')).toBeInTheDocument();
+    expect(apiClientOverrides.apiCreateSetupIntent).toHaveBeenCalledTimes(1);
+    expect(stripeOverride.confirmCardSetup).toHaveBeenCalledWith(
+      ...EXPECTED_CONFIRM_CARD_SETUP_ARGS
+    );
+    expect(
+      apiClientOverrides.apiUpdateDefaultPaymentMethod
+    ).not.toHaveBeenCalled();
+    expect(refreshSubscriptions).not.toHaveBeenCalled();
+  });
+
+  it('displays an error if confirmCardSetup returns a setupIntent without a payment method ID', async () => {
+    const {
+      apiClientOverrides,
+      stripeOverride,
+      refreshSubscriptions,
+    } = await commonSubmitSetup({
+      stripeOverride: {
+        ...defaultStripeOverride(),
+        confirmCardSetup: jest
+          .fn()
+          .mockResolvedValue({ setupIntent: { payment_method: false } }),
+      },
+    });
+
+    expect(screen.queryByTestId('error-message-container')).toBeInTheDocument();
+    expect(screen.getByText('basic-error-message')).toBeInTheDocument();
+    expect(screen.queryByTestId('paymentForm')).toBeInTheDocument();
+    expect(apiClientOverrides.apiCreateSetupIntent).toHaveBeenCalledTimes(1);
+    expect(stripeOverride.confirmCardSetup).toHaveBeenCalledWith(
+      ...EXPECTED_CONFIRM_CARD_SETUP_ARGS
+    );
+    expect(
+      apiClientOverrides.apiUpdateDefaultPaymentMethod
+    ).not.toHaveBeenCalled();
+    expect(refreshSubscriptions).not.toHaveBeenCalled();
+  });
+
+  it('displays an error if apiUpdateDefaultPaymentMethod throws', async () => {
+    const {
+      apiClientOverrides,
+      stripeOverride,
+      refreshSubscriptions,
+    } = await commonSubmitSetup({
+      apiClientOverrides: {
+        ...defaultApiClientOverrides(),
+        apiUpdateDefaultPaymentMethod: jest
+          .fn()
+          .mockRejectedValue('barf apiUpdateDefaultPaymentMethod'),
+      },
+    });
+
+    expect(screen.queryByTestId('error-message-container')).toBeInTheDocument();
+    expect(screen.getByText('basic-error-message')).toBeInTheDocument();
+    expect(screen.queryByTestId('paymentForm')).toBeInTheDocument();
+    expect(apiClientOverrides.apiCreateSetupIntent).toHaveBeenCalledTimes(1);
+    expect(stripeOverride.confirmCardSetup).toHaveBeenCalledWith(
+      ...EXPECTED_CONFIRM_CARD_SETUP_ARGS
+    );
+    expect(
+      apiClientOverrides.apiUpdateDefaultPaymentMethod
+    ).toHaveBeenCalledWith({
+      paymentMethodId: CONFIRM_CARD_SETUP_RESULT.setupIntent.payment_method,
+    });
+    expect(refreshSubscriptions).not.toHaveBeenCalled();
+  });
+
+  describe('Localized Plan Billing Description Component', () => {
```

Comment:
All of these l10n tests in the second half of this new `PaymentUpdateFormV2.test.tsx` are copied verbatim from `PaymentUpdateForm.test.tsx`